src/platform/testing/contract/index.js

```diff
@@ -7,22 +7,38 @@ import { Pact } from '@pact-foundation/pact';
  * @param {string} consumer - String label for the consumer.
  * @param {string} provider - String label for the provider.
  * @param {function} test - Callback function for a test suite (describe block).
+ *                          Receives one argument: a function that returns the
+ *                          mock provider instance for the test.
  */
 const contractTest = (consumer, provider, test) => {
   describe(consumer, () => {
-    const mockApi = new Pact({
-      port: 3000,
-      consumer,
-      provider,
-      spec: 2,
-      cors: true,
+    // The mock server is initialized in the before hook so that
+    // separate contract tests don't conflict by attempting to
+    // create mock servers at the same port.
+    let mockApi;
+
+    // This wrapper function is used as a closure around mockApi
+    // to indirectly pass the mock server to the test callback
+    // since it's not initialized until the before hook runs.
+    const mockApiWrapper = () => mockApi;
+
+    before('set up mock server', () => {
+      mockApi = new Pact({
+        port: 3000,
+        consumer,
+        provider,
+        spec: 2,
+        cors: true,
+      });
+
+      return mockApi.setup();
     });
 
-    before(() => mockApi.setup());
-    after(() => mockApi.finalize());
-    afterEach(() => mockApi.verify());
+    after('generate pacts', () => mockApi.finalize());
+
+    afterEach('verify interactions', () => mockApi.verify());
```

Comment:
Added [descriptions for hooks](https://mochajs.org/#describing-hooks) to improve debugging.